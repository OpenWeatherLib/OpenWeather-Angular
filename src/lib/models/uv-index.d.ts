export interface UvIndex {
  lat: number;
  lon: number;
  date_iso: Date;
  date: Date;
  value: number;
}
