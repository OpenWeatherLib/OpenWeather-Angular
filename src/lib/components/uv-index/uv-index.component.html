<mat-card class="ga-card" *ngIf="uvIndex">
  <mat-card-header>
    <img mat-card-avatar class="ga-card-header-image" src="/assets/uv_index.png" alt="UvIndex pic">
    <mat-card-title>UvIndex</mat-card-title>
    <mat-card-subtitle>{{uvIndex?.value}}</mat-card-subtitle>
  </mat-card-header>
  <mat-card-content>
    <p>Coordinates: Lat: {{uvIndex?.lat}}, Lon: {{uvIndex?.lon}}</p>
  </mat-card-content>
</mat-card>
