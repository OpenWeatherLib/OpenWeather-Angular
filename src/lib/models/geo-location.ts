import { classJsonKey } from "../decorator";

@classJsonKey("", "")
export class GeoLocation {
  lat: number = 720.0;
  long: number = 720.0;

  isDefault(): boolean {
    return this.lat === 720.0 && this.long === 720.0;
  }
}
