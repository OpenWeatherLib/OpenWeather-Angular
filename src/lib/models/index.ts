export * from "./address-component";
export * from "./city";
export * from "./city2";
export * from "./clouds";
export * from "./coordinates";
export * from "./coordinates2";
export * from "./geometry";
export * from "./main";
export * from "./rain";
export * from "./sys";
export * from "./weather-current";
export * from "./weather-forecast";
export * from "./weather-forecast-part";
export * from "./weather-part";
export * from "./wind";
export * from "./uv-index";
