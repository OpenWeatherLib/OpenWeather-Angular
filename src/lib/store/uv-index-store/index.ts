export * from "./uv-index.actions";
export * from "./uv-index.effects";
export * from "./uv-index.reducer";
export * from "./uv-index.selectors";
export * from "./uv-index.state";
