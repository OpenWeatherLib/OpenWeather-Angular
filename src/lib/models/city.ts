import { GeoLocation } from "./geo-location";
import { classJsonKey } from "../decorator";

@classJsonKey("", "")
export class City {
  id: number = 0;
  name: string = "";
  country: string = "";
  population: number = 0;
  geoLocation: GeoLocation = new GeoLocation();

  isDefault(): boolean {
    return this.id === 0 && this.name === "" && this.country === "" && this.population === 0 && this.geoLocation.isDefault();
  }
}
