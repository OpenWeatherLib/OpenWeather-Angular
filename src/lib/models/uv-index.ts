import { GeoLocation } from "./geo-location";

export class UvIndex {
  geoLocation: GeoLocation = new GeoLocation();
  date: Date = new Date();
  value: number = 0.0;

  isDefault(): boolean {
    return this.geoLocation.isDefault() && this.value === 0.0;
  }
}
