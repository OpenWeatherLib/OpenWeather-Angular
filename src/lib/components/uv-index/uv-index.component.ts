import { Component, OnInit } from "@angular/core";
import { Store } from "@ngrx/store";
import { Observable } from "rxjs";

import { UvIndex } from "@lib/models";
import { selectIsLoading, selectUvIndex } from "@lib/store/uv-index-store";
import { RootState } from "@lib/store/root-state";

@Component({
  selector: "ga-uv-index",
  templateUrl: "./uv-index.component.html",
  styleUrls: ["./uv-index.component.scss"]
})
export class UvIndexComponent implements OnInit {

  isLoading$: Observable<boolean>;

  uvIndex$: Observable<UvIndex>;

  constructor(private readonly store$: Store<RootState>) { }

  ngOnInit() {
    this.isLoading$ = this.store$.select(selectIsLoading);

    this.uvIndex$ = this.store$.select(selectUvIndex);
  }
}
