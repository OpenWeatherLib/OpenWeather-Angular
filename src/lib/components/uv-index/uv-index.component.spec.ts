/*eslint-disable @typescript-eslint/camelcase */

import { NO_ERRORS_SCHEMA } from "@angular/core";
import { TestBed, async } from "@angular/core/testing";
import { combineReducers, Store, StoreModule } from "@ngrx/store";
import { of } from "rxjs";

import { UvIndex } from "@lib/models";
import { uvIndexReducer } from "@lib/store/uv-index-store";
import { RootState } from "@lib/store/root-state";
import { UvIndexComponent } from "./uv-index.component";

describe("UvIndexComponent", () => {
  let component: UvIndexComponent;
  let store$: Store<RootState>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        UvIndexComponent
      ],
      imports: [
        StoreModule.forRoot({ uvIndex: combineReducers(uvIndexReducer) })
      ],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();

    store$ = TestBed.get(Store);
    store$.select = jest.fn()
      .mockReturnValueOnce(of(false))
      .mockReturnValueOnce(of({
        lat: 38.75,
        lon: 40.25,
        date_iso: "2017-06-23T12:00:00Z",
        date: 1498219200,
        value: 10.16
      }));

    const fixture = TestBed.createComponent(UvIndexComponent);
    component = fixture.debugElement.componentInstance;
  });

  test("should create the app", async(() => {
    // Assert
    expect(component).toBeTruthy();
  }));

  describe("ngOnInit", () => {
    test("isLoading$", done => {
      // Arrange 
      component.ngOnInit();

      // Act
      component.isLoading$
        .subscribe({
          next: (isLoading: boolean) => {
            // Assert
            expect(isLoading).toBeFalsy();
          },
          complete: () => done()
        });
    });

    test("uvIndex$", done => {
      // Arrange 
      component.ngOnInit();

      // Act
      component.uvIndex$
        .subscribe({
          next: (uvIndex: UvIndex) => {
            // Assert
            expect(uvIndex).toMatchSnapshot({
              lat: 38.75,
              lon: 40.25,
              date_iso: "2017-06-23T12:00:00Z",
              date: 1498219200,
              value: 10.16
            });
          },
          complete: () => done()
        });
    });
  });
});
