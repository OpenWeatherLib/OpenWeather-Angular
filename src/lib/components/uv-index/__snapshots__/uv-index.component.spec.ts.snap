// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UvIndexComponent ngOnInit uvIndex$ 1`] = `
Object {
  "date": 1498219200,
  "date_iso": "2017-06-23T12:00:00Z",
  "lat": 38.75,
  "lon": 40.25,
  "value": 10.16,
}
`;
